server/segment_rpc.go

```diff
@@ -46,14 +45,15 @@ var errCapCompat = errors.New("incompatible capabilities")
 
 var tlsConfig = &tls.Config{InsecureSkipVerify: true}
 var httpClient = &http.Client{
-	Transport: &http2.Transport{
+	Transport: &http.Transport{
 		TLSClientConfig: tlsConfig,
-		DialTLS: func(network, addr string, cfg *tls.Config) (gonet.Conn, error) {
+		DialTLSContext: func(ctx context.Context, network, addr string) (gonet.Conn, error) {
 			netDialer := &gonet.Dialer{
 				Timeout: 2 * time.Second,
 			}
-			return tls.DialWithDialer(netDialer, network, addr, cfg)
+			return tls.DialWithDialer(netDialer, network, addr, tlsConfig)
 		},
+		ForceAttemptHTTP2: true,
```

Comment:
According to the [http.Transport docs](https://pkg.go.dev/net/http#Transport) and [the http package code](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/net/http/transport.go;l=387), `ForceAttemptHTTP2: true` is needed for the transport to try to upgrade to HTTP/2 if `TLSClientConfig` is non-nil or if custom dialers (i.e. `DialTLSContext`) are used. By including this, we can default to transparently supporting HTTP/2 while maintaining the flexibility to use HTTP/1 by running with `GODEBUG=http2client=0`.

Reply:
Nit: You can add a comment into the code for the explanation.

Reply:
Fixed in ca94081